Create new layer for "ref" of minor roads and paths

jeisenbe opened the pull request:
Partially replaces PR #3663
Follow-up to PR #3654

### Changes proposed in this pull request:

1) Move unclassified, residential and track highways, and aeroways "ref" to a later layer, after name labels are rendered

2) Change text-halo to standard size for highway=unclassified/residential refs and aeroway refs

3) Change residential / unclassified road "ref" size to 11 point at z17, to be the same size as the name text labels at this zoom level and higher. 

### Explanation

**1) The minor roads, aeroways, and tracks with text-based ref labels currently render in the same layer as the shields for major roads.** 

- The ref text is rendered before the name label, and can block the name from rendering when spaces is limited.
- Major road shields render much sooner than the ref text labels of minor roads, which are not shown until z15 or z16.
- Major road shields are usually more important than the name; the number of a motorway is used more often, but for minor roads, such as residential streets, the name is more useful and should be shown with higher priority
- Changing residential, unclassified and tracks to a later layer will fix this rendering. Aeroways "ref" text labels are also moved to this new layer because they are also rendered only at z15 and higher, though this does not change the cartography because they do not have a rendering for names

**2) The "ref" labels rendering for residential/unclassified roads needs minor adjustments**
 At z17, the "name" text label increases to 11pt font size, but the ref does not change size till z18 currently. This should be adjusted so that the "ref" label is the same size as the name label, as at z15 and z16.

**3) The "ref" halo needs to be changed to standard size.**
 Right now it is 2 pixels wide, which would overlap with the casing at certain zoom levels. 

### Test rendering with links to the example places:

**highway=unclassified with name and ref** - Basilicata, Italy 
https://www.openstreetmap.org/#map=17/40.3776321/15.7281581
Before z17
![via-chiusululle-unclassified-master](https://user-images.githubusercontent.com/42757252/52036343-55596e80-2570-11e9-92ba-d9ae1d158950.png)
z18
![z18-via-chiusululle-master](https://user-images.githubusercontent.com/42757252/52036353-5b4f4f80-2570-11e9-8191-8d5a7b4ec879.png)
After z17
![chiusululle-z17-after](https://user-images.githubusercontent.com/42757252/52040702-a0c54a00-257b-11e9-9b31-d594436d84b5.png)
z18
![chiusululle-z18-after](https://user-images.githubusercontent.com/42757252/52040720-a6bb2b00-257b-11e9-9baa-e369050e3a22.png)


**highway=residential with name and ref** - Basilicata, Italy 
https://www.openstreetmap.org/#map=17/40.1412792/16.0893142
Before z17
![via-vittorio-res-z17-master](https://user-images.githubusercontent.com/42757252/52036418-74f09700-2570-11e9-9150-b7f89441fcb0.png)
z18
![via-vitorio-z18-master](https://user-images.githubusercontent.com/42757252/52036426-77eb8780-2570-11e9-9e24-bddfaf3357c9.png)
After z17
![z17-via-vittorio-after](https://user-images.githubusercontent.com/42757252/52036435-7cb03b80-2570-11e9-9cd9-256760bfc710.png)
z18
![via-vittori-z18-after](https://user-images.githubusercontent.com/42757252/52040731-ae7acf80-257b-11e9-8a78-e0b4b13777dc.png)

jeisenbe commented:
Does anyone have time to look at this PR?

kocio-pl commented:
I don't have much time lately, but looking at the rendering this is a good idea.

imagico commented:
I am sorry for the long wait - I looked at this and am concerned about two things:

* by moving aeroway refs to the new layer you make them lower priority than road names.  I am not sure this would work so well - in particular for service roads and tunnels on airports
* rendering refs with labels indistinguishable from names does not seem a good idea.

jeisenbe commented:
> * by moving aeroway refs to the new layer you make them lower priority than road names.  I am not sure this would work so well - in particular for service roads and tunnels on airports

The "ref" for runways (and taxiways) is only rendered at z15 and greater, and it's rendered one time at the center of the way. So this is only an issue for airports where a named road crosses the center point a runway or taxiway. I can think of several examples of paths and roads that cross over or level with runways (eg Gibraltar, LAX, some airstrips in my area) but I don't know of any where the road is close enough to the runway center for this to cause interference with the label, so I doubt this is every an issue in practice. 

However, It would be easy enough to keep the aeroways in the main `roads-text-ref` layer. I only moved them because they render at z15 and higher only, so I thought it would slightly improve rendering performance for z13 and z14, which are included in `roads-text-ref` but not in the new layer.

> * rendering refs with labels indistinguishable from names does not seem a good idea.

This is the current situation with "ref" on residential, unclassified and track roads. The PR only makes 2 tiny adjustments for consistency:

```
-      [zoom >= 18] {
-        text-size: 10;
+      [zoom >= 17] {
+        text-size: 11;
```

```
-      text-halo-radius: 2;
+      text-halo-radius: @standard-halo-radius;
```

If we want the 'ref' to be clearly distinct from the road name label on this minor roads, there would need to be a more significant distinction than the halo radius or a 1 point font size difference at z18 and z19. 

I could try rendering the ref with italic font, perhaps? Or should this be a separate PR?

jeisenbe commented:
Does anyone else have comments about this? Should I move the aeroways back to the earlier layer? Does the "ref" label need to be more clearly distinguished from the "name" label?

jeisenbe commented:
1) Do we need to keep  aeroways in the earlier layer? 
2) Should the "ref" rendering be changed in this PR, or can it wait till later?
See comments above https://github.com/gravitystorm/openstreetmap-carto/pull/3709#issuecomment-478300867

imagico commented:
Would be good to have another opinion on this, maybe from @matkoniecz, @kocio-pl or @sommerluk.  I mentioned two points of concern, but these are not essential for me.  Another look at this from someone else would be useful.

sommerluk commented:
Okay. I didn’t look to the code. But the idea to unify the `ref` label rendering with the `name` label rendering seems good to me. The current situation is that we have some differences in font size between `name` and `ref`, and these differences are not big enough within the same zoom level (and not systematic enough across different zoom levels) to allow a clear distinction anyway. Personally I also think different font sizes on the same zoom level for the same object in the same rendering (here: along the line) make the map harder to read (less intuitive).

Using the italic font instead of the regluar font for `ref` might also work. (Side note: We have currently no systematic approach about when using italic fonts and when not.)

jeisenbe commented:
@sommerluk - any thoughts on including aeroway refs in this new layer
or not? I'm happy to change it, but I hoped it could be slightly
better performance this way.

jeisenbe commented:
I've moved aeroway ref rendering back to the original layer (roads-text-ref)
I also changed the rendering of minor roads and aeroways to use an oblique (italic) font, as suggested above

jeisenbe commented:
Test renderings with oblique font for ref. It's a subtle change:

highway=unclassified with name and ref - Basilicata, Italy
https://www.openstreetmap.org/#map=17/40.3776321/15.7281581
![z18-ref-oblique-via-chiusululle](https://user-images.githubusercontent.com/42757252/58058485-24935880-7ba4-11e9-98f8-ab0978784fcc.png)

highway=residential with name and ref - Basilicata, Italy
https://www.openstreetmap.org/#map=17/40.1412792/16.0893142
![z18-ref-oblique-via-vittorio](https://user-images.githubusercontent.com/42757252/58058492-2b21d000-7ba4-11e9-8a59-4119909b533b.png)
![z17-ref-oblique-via-vittorio](https://user-images.githubusercontent.com/42757252/58058498-2fe68400-7ba4-11e9-97af-a2a881a15ea7.png)

Runway with ref in new oblique font:
![z15-wamena-runway-15-33](https://user-images.githubusercontent.com/42757252/58062206-6deba400-7bb4-11e9-9155-9e5617ea88f6.png)

(Compare to current rendering:)
![z15-current-runway-rendering](https://user-images.githubusercontent.com/42757252/58062216-747a1b80-7bb4-11e9-9b5c-179f5ae196f6.png)

jeisenbe commented:
Any comments on the new commit?

jeisenbe commented:
@imagico, your original 2 points of concern were addressed by the most recent commits.

jeisenbe commented:
@imagico has approved this now, and it looks like @kocio-pl and @sommerluk were in favor, so I'm going to merge this PR tomorrow, once I figure out when the changelog should be updated.

pnorman commented:
> once I figure out when the changelog should be updated

It's not necessary, but a big help when releasing. Just edit the changelog directly on master.
